types/relay-compiler/lib/core/GraphQLIRVisitor.d.ts

```diff
@@ -0,0 +1,109 @@
+import {
+    Argument,
+    ClientExtension,
+    Condition,
+    Defer,
+    ConnectionField,
+    Directive,
+    Fragment,
+    FragmentSpread,
+    InlineDataFragmentSpread,
+    InlineFragment,
+    LinkedField,
+    Literal,
+    LocalArgumentDefinition,
+    ModuleImport,
+    Request,
+    Root,
+    RootArgumentDefinition,
+    ScalarField,
+    SplitOperation,
+    Stream,
+    Variable,
+} from './GraphQLIR';
+
+export type VisitNode =
+    | Argument
+    | ClientExtension
+    | Condition
+    | Defer
+    | ConnectionField
+    | Directive
+    | Fragment
+    | FragmentSpread
+    | InlineDataFragmentSpread
+    | InlineFragment
+    | LinkedField
+    | Literal
+    | LocalArgumentDefinition
+    | ModuleImport
+    | Request
+    | Root
+    | RootArgumentDefinition
+    | ScalarField
+    | SplitOperation
+    | Stream
+    | Variable;
+
+type EnterLeave<T> = { readonly enter?: T; readonly leave?: T };
+
+export type VisitFn<T extends VisitNode> = (
+    node: T, // node we're visiting
+    key?: any, // index/key to node from parent array/object
```

Comment:
I would prefer `unknown`

Reply:
Hmm, yeah I get that it would normally be the preferable type, but this is a straight port [from upstream](https://github.com/facebook/relay/blob/v6.0.0/packages/relay-compiler/core/GraphQLIRVisitor.js#L90) where they also went with `any`. I guess I prefer to keep things easily portable, but don’t feel strongly about that, so if you do then I'll update this 👍

Reply:
Hm then it might be okay. How is any from flow different than typescripts any?

Reply:
Afaik they are pretty much equivalent. Flow’s `mixed` is similar to TS’ `unknown`.

Reply:
Alright!